arch/x86/adler32_avx512.c

```diff
@@ -15,13 +15,32 @@
 
 #ifdef X86_AVX512_ADLER32
 
+#include <cpu_features.h>
+
 Z_INTERNAL uint32_t adler32_avx512(uint32_t adler, const unsigned char *buf, size_t len) {
     uint32_t sum2;
 
+    /* For impossibly tiny sizes, use the smaller width versions. We still need
+     * to check for compile time support for these but they are likely there */
+#ifdef X86_SSE41_ADLER32 
+    if (len < 32) {
+       return adler32_sse41(adler, buf, len); 
+    }
+#endif
+
+#ifdef X86_AVX2_ADLER32
+    if (len < 64) {
```

Comment:
Looks like there was two places with the extra braces. You got the one I didn't mention below.